### PR TITLE
Integration Test : JTA Service Migration 

### PR DESCRIPTION
**Pull request by anpanigr**
The PR adds the following usecase 

- Verify JTA Recovery Service is migrated to an available active server when a server is shutdown ( cluster is scaled down )
- The MigrationPolicy on the ServerTemplate is set to 'shutdown-recovery' so that the JTA recovery service is migrated to an active server in the cluster when a server is shutdown. 
- This can be checked by verifying the JTARecoveryService runtime MBean for the stopped server in an active server.  For example say manged server ms2 is down, make sure that the JTA recovery service for ms2 is active on the running managed server ms1
- Make sure the JTA recovery service is migrated back to original hosting server when it is brought back to running state.

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4842/  (all pass)

### CODE REVIEW

**Review by ajsomogyi — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 1]
```diff
@@ -351,4 +399,40 @@ private boolean checkJmsServerRuntime(String managedServer) {
     return true;
   }
 
+  /*
+   * Verify the JTA Recovery Service Runtime through rest API.
```
Comment (marinakog) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:403]: REST API

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 2]
```diff
@@ -351,4 +399,40 @@ private boolean checkJmsServerRuntime(String managedServer) {
...
+   * Get the JTA Recovery Service Runtime for a server on a 
+   * specified managed server.
+   * @param managedServer name of the server to look for RecoveyServerRuntime
+   * @param recoveryService name of RecoveyServerRuntime (managed server) 
+   * @param active is the recovery active (true or false )
+   * @returns true if MBEAN is found otherwise false
```
Comment (marinakog) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:409]: MBean

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 3]
```diff
@@ -351,4 +399,40 @@ private boolean checkJmsServerRuntime(String managedServer) {
...
+   **/
+  private boolean checkJtaRecoveryServiceRuntime(String managedServer, String recoveryService, String active) {
+    ExecResult result = null;
+    int adminServiceNodePort
+        = getServiceNodePort(domainNamespace, getExternalServicePodName(adminServerPodName), "default");
+    StringBuffer curlString = new StringBuffer("curl --user weblogic:welcome1 ");
+    curlString.append("\"http://" + K8S_NODEPORT_HOST + ":" + adminServiceNodePort)
```
Comment (marinakog) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:416]: hard coded values - switch to curl --user " + ADMIN_USERNAME_DEFAULT + ":" + ADMIN_PASSWORD_DEFAULT

**Review by marinakog — CHANGES_REQUESTED** [review]

**Review by vanajamukkara — APPROVED** [review]
lg

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java (hunk 3 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:418]: done

**Review by anpanigr — COMMENTED** [review]
done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java (hunk 1 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java (hunk 2 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java]: done

**Review by marinakog — APPROVED** [review]

**Pull request merged**